source/extensions/quic_listeners/quiche/platform/envoy_quic_clock.cc

```diff
@@ -0,0 +1,21 @@
+#include "extensions/quic_listeners/quiche/platform/envoy_quic_clock.h"
+
+namespace Envoy {
+namespace Quic {
+
+quic::QuicTime EnvoyQuicClock::ApproximateNow() const {
+  // This might be expensive as Dispatcher doesn't store approximate time_point.
+  return Now();
+}
+
+quic::QuicTime EnvoyQuicClock::Now() const {
+  return quic::QuicTime::Zero() +
+         quic::QuicTime::Delta::FromMicroseconds(timePointToInt64(time_system_.monotonicTime()));
+}
+
+quic::QuicWallTime EnvoyQuicClock::WallNow() const {
+  return quic::QuicWallTime::FromUNIXMicroseconds(timePointToInt64(time_system_.systemTime()));
```

Comment:
The standard says std::chrono::system_clock, the source of systemTime(), has a unspecified epoch, although
-  Most implementations already uses Unix epoch.
-  It will be Unix epoch from c++ 20.

So I guess this is fine in practice?